packages/datastore/src/util.ts

```diff
@@ -681,3 +682,71 @@ export const isAWSIPAddress = (val: string): boolean => {
 		val
 	);
 };
+
+export function DeferredPromise(): void {
+	const self = this;
+	this.promise = new Promise((resolve, reject) => {
+		self.resolve = resolve;
+		self.reject = reject;
+	});
+}
+
+export class SubscriptionBuffer {
+	private limitPromise = new DeferredPromise();
+	private timerPromise: Promise<string>;
+	private TIMER_INTERVAL = 2000;
+	private timer: NodeJS.Timer;
```

Comment:
It feels weird to have a `NodeJS` type here. Do you think it's better to use `ReturnType<typeof setTimeout>` instead? 😄 This way we can just let Typescript infer the type for us.
```suggestion
	private timer: ReturnType<typeof setTimeout>;
```